src/bootstrap/bootstrap.py

```diff
@@ -1162,6 +1162,43 @@ def build_triple(self):
         config = self.get_toml("build")
         return config or default_build_triple(self.verbose)
 
+    def is_git_repository(self, repo_path):
+        return os.path.isdir(os.path.join(repo_path, ".git"))
+
+    def get_latest_commit(self, repo_path, author_email):
+        try:
+            if (
+                not repo_path
+                or not author_email
```

Comment:
Can these ever be empty?

Reply:
Can you guess the reason? They should never be empty actually.

Reply:
`stage0` file always exists so I expect that logic to work properly without `if not repo_path or not author_email` condition unless we have a bug or something..